File: vito/core/src/main/java/com/facebook/fresco/vito/core/DefaultFrescoVitoConfig.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.fresco.vito.core;

import com.facebook.common.internal.Supplier;
import com.facebook.common.internal.Suppliers;
import com.facebook.infer.annotation.Nullsafe;
import com.facebook.infer.annotation.OkToExtend;

@Nullsafe(Nullsafe.Mode.STRICT)
@OkToExtend
public class DefaultFrescoVitoConfig implements FrescoVitoConfig {

  private final PrefetchConfig mPrefetchConfig;

  public DefaultFrescoVitoConfig() {
    this(new DefaultPrefetchConfig());
  }

  public DefaultFrescoVitoConfig(PrefetchConfig prefetchConfig) {
    mPrefetchConfig = prefetchConfig;
  }

  @Override
  public PrefetchConfig getPrefetchConfig() {
    return mPrefetchConfig;
  }

  @Override
  public boolean submitFetchOnBgThread() {
    return true;
  }

  @Override
  public boolean useBindOnly() {
    return false;
  }

  @Override
  public boolean useNewReleaseCallback() {
    return false;
  }

  @Override
  public Supplier<Boolean> useNativeRounding() {
    return Suppliers.BOOLEAN_TRUE;
  }

  @Override
  public boolean layoutPrefetchingEnabled(Object surface) {
    return false;
  }

  public static class DefaultPrefetchConfig implements PrefetchConfig {
    @Override
    public boolean prefetchInOnPrepare() {
      return true;
    }

    @Override
    public PrefetchTarget prefetchTargetOnPrepare() {
      return PrefetchTarget.MEMORY_DECODED;
    }

    @Override
    public boolean cancelOnPreparePrefetchWhenWorkingRangePrefetch() {
      return true;
    }

    @Override
    public boolean cancelPrefetchWhenFetched() {
      return true;
    }

    @Override
    public boolean prefetchWithWorkingRange() {
      return true;
    }

    @Override
    public int prefetchWorkingRangeSize() {
      return 3;
    }

    @Override
    public PrefetchTarget prefetchTargetWorkingRange() {
      return PrefetchTarget.MEMORY_DECODED;
    }
  }
}

File: vito/core/src/main/java/com/facebook/fresco/vito/core/DefaultFrescoVitoConfig.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.fresco.vito.core

import com.facebook.common.callercontext.ContextChain
import com.facebook.common.internal.Supplier
import com.facebook.common.internal.Suppliers

open class DefaultFrescoVitoConfig
@JvmOverloads
constructor(override val prefetchConfig: PrefetchConfig = DefaultPrefetchConfig()) :
    FrescoVitoConfig {

  override fun submitFetchOnBgThread(): Boolean = true

  override fun useBindOnly(): Boolean = false

  override fun useNewReleaseCallback(): Boolean = false

  override fun useNativeRounding(): Supplier<Boolean>? = Suppliers.BOOLEAN_TRUE

  override fun layoutPrefetchingEnabled(callerContext: Any?): Boolean = false

  override fun useSmartPropertyDiffing(): Boolean = false

  override fun onlyStopAnimationWhenAutoPlayEnabled(): Boolean = true

  override fun fastPathForEmptyRequests(): Boolean = false

  override fun enableWindowWideColorGamut(): Boolean = false

  override fun handleImageResultInBackground(): Boolean = false

  override fun useIntermediateImagesAsPlaceholder(): Boolean = false

  override fun fallbackToDefaultImageOptions(): Boolean = false

  override fun experimentalDynamicSizeVito2(): Boolean = false

  override fun experimentalDynamicSizeWithCacheFallbackVito2(): Boolean = false

  override fun experimentalDynamicSizeOnPrepareMainThreadVito2(): Boolean = false

  override fun experimentalDynamicSizeDiskCacheCheckTimeoutMs(): Long = 0

  override fun experimentalDynamicSizeUseSfOnDiskCacheTimeout(): Boolean = false

  override fun isAppStarting(): Boolean = false

  override fun experimentalDynamicSizeDisableWhenAppIsStarting(): Boolean = false

  override fun experimentalDynamicSizeCheckIfProductIsEnabled(): Boolean = false

  override fun experimentalDynamicSizeIsProductEnabled(
      callerContext: Any?,
      contextChain: ContextChain?
  ): Boolean = true

  override fun experimentalResetVitoImageRequestListener() = false

  override fun experimentalResetLocalVitoImageRequestListener() = false

  override fun experimentalResetLocalImagePerfStateListener() = false

  override fun experimentalResetControllerListener2() = false

  open class DefaultPrefetchConfig : PrefetchConfig {
    override fun prefetchInOnPrepare(): Boolean = true

    override fun prefetchInOnBoundsDefinedForDynamicSize(): Boolean = false

    override fun prefetchTargetOnPrepare(): PrefetchTarget = PrefetchTarget.MEMORY_DECODED

    override fun prefetchTargetOnBoundsDefined(): PrefetchTarget = PrefetchTarget.MEMORY_DECODED
  }
}
